client/src/components/chat/ChatWindow.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react"
+import Message from "./Message"
+
+const ChatWindow = ({ user, roomId, messages, sendMessage }) => {
+  const [newMessage, setNewMessage] = useState({
+    text: "",
+    user: {
+      id: user.id,
+      email: user.email
+    },
+    room: roomId
```

Comment:
shouldnt cause issues, but is there a reason why roomId is in state?

Reply:
SocketIO on the back end looks at what room the message is associated with to determine where to broadcast it